hs_core/hydroshare/resource.py

```diff
@@ -411,24 +405,18 @@ def create_resource(
             k, v = element.items()[0]
             resource.metadata.create_element(k, **v)
 
-        # add the subject elements from the AssignedKeywords (new metadata implementation)
-        for akw in AssignedKeyword.objects.filter(object_pk=resource.id).all():
-            resource.metadata.create_element('subject', value=akw.keyword.title)
+        for keyword in keywords:
+            resource.metadata.create_element('subject', value=keyword)
 
         hs_bagit.create_bag(resource)
 
-    except Exception as ex:
-        if resource:
-            resource.delete()
```

Comment:
This resource.delete() not only delete the resource table in the database, but also delete the resource files and bags from iRODS storage accordingly. See delete() method on AbstractResource class in hs_core/models.py before the refactoring. So perhaps this exception handling is still needed to roll back/delete the resource storage on iRODS side?